homeassistant/components/automation/time.py

```diff
@@ -13,20 +13,29 @@
 _LOGGER = logging.getLogger(__name__)
 
 TRIGGER_SCHEMA = vol.Schema(
-    {vol.Required(CONF_PLATFORM): "time", vol.Required(CONF_AT): cv.time}
+    {
+        vol.Required(CONF_PLATFORM): "time",
+        vol.Required(CONF_AT): vol.All(cv.ensure_list, [cv.time]),
+    }
 )
 
 
 async def async_attach_trigger(hass, config, action, automation_info):
     """Listen for state changes based on configuration."""
-    at_time = config.get(CONF_AT)
-    hours, minutes, seconds = at_time.hour, at_time.minute, at_time.second
+    at_times = config[CONF_AT]
 
     @callback
     def time_automation_listener(now):
         """Listen for time changes and calls action."""
         hass.async_run_job(action, {"trigger": {"platform": "time", "now": now}})
 
-    return async_track_time_change(
-        hass, time_automation_listener, hour=hours, minute=minutes, second=seconds
-    )
+    return [
```

Comment:
We shouldn't return a list. Instead we should just return a function that iterates over the list of results.

Reply:
Done